app/models/device.rb

```diff
@@ -0,0 +1,80 @@
+class Device < ApplicationRecord
+  belongs_to :user
+
+  IOS = "iOS".freeze
+  ANDROID = "Android".freeze
+
+  validates :token, uniqueness: { scope: %i[user_id platform app_bundle] }
+  validates :platform, inclusion: { in: %w[iOS Android] }
```

Comment:
Nitpicky: Should this be using the constants too?

```suggestion
  validates :platform, inclusion: { in: [IOS, ANDROID] }
```

Reply:
Yes, thanks for the catch!